### PR TITLE
Added workshop instructions (except ChatGPT).

### PR DESCRIPTION
**Pull request by mwhittaker**
**REVIEWERS**: See https://github.com/ServiceWeaver/workshops/tree/emoji_instructions for a rendered version of this PR.

This PR adds the instructions that participants will follow (except for the ChatGPT part which I'll finish in a future PR). Once we finalize the instructions, I'll split the solutions into the various directories and embed references to them in the instructions. I'll also add a link to a Google Form where people can leave feedback.

### CODE REVIEW

File: 07/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+["☘️","🚀","🪨"]
+$ curl localhost:9000/search?q=rock
+["☘️","🚀","🪨"]
+$ curl localhost:9000/search?q=rock
+["☘️","🚀","🪨"]
+$ curl localhost:9000/search?q=rock
+["☘️","🚀","🪨"]
+```
+
+Surprisingly, every request is slow! Add some logs to the `Get` and `Put`
```
Comment (spetrovic77) [inline 07/README.md:88]: This was indeed surprising, even to me!

File: 08/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+Build and run your application using `weaver multi deploy`:
+
+```
+$ weaver generate .
+$ go build .
+$ weaver multi deploy config.toml
+```
+
+And again in a separate terminal, repeatedly curl the application for the prime
```
Comment (spetrovic77) [inline 08/README.md:32]: This is stale 😆

File: 08/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+["☘️","🚀","🪨"]
+$ curl localhost:9000/search?q=rock
+["☘️","🚀","🪨"]
+$ curl localhost:9000/search?q=rock
+["☘️","🚀","🪨"]
+```
+
+The first request should be slow, but all subsequent requests should complete
+nearly instantly. If you look at your application logs, you can confirm that
+`Get` and `Put` requests for the same query are routed to the same `Cache`
```
Comment (spetrovic77) [inline 08/README.md:48]: s/the same query/the query "rock"/

**Review by spetrovic77 — APPROVED** [review]
Looks awesome, really great work!

File: 02/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+```go
+Search(context context.Context, query string) ([]string, error)
+```
+
+The `Search` method receives a query like `"black cat"` and returns the emojis
+that match the query. To implement the `Search` method, first lowercase the
+query ([`strings.ToLower`](https://pkg.go.dev/strings#ToLower)) and then split
+the query into words ([`strings.Fields`](https://pkg.go.dev/strings#Fields)).
+Then iterate over the `emojis` map in `emojis.go` to find all matching emojis.
+We say an emoji *matches* a query if every word in the query is a substring of
```
[the remark this answers is not in the document]
Comment (mwhittaker) [reply 02/README.md]: D'oh, you're totally right. Updated the code and examples.

File: 02/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
+# A Search Component
+
+In this part, you will implement a simple component that searches for emojis
+that match a query. First, download or copy-and-paste [`emojis.go`](emojis.go)
+into your `emojis/` directory. This file contains a `map[string][]string` called
+`emojis` that maps every emoji to a list of labels. For example, the black cat
+emoji 🐈‍⬛ has labels `"animal"`, `"animals"`, `"black"`, `"cat"`,
+`"mammal"`, and `"nature"`. Refer to github.com/mwhittaker/emojis if you want to
```
[the remark this answers is not in the document]
Comment (mwhittaker) [reply 02/README.md]: Done.

File: 04/README.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+<details>
+<summary>Solution.</summary>
+
+TODO(mwhittaker): Embed solution here.
+</details>
+
+Next, inside of the `Main` method, implement an HTTP handler for the `/search`
+endpoint. The endpoint receives GET requests of the form `/search?q=<query>` and
+returns the JSON serialized list of emojis that match the query. For example,
+`curl localhost:9000/search?q=rock` should return `["☘️","🚀","🪨"]`. At the end
```
[the remark this answers is not in the document]
Comment (mwhittaker) [reply 04/README.md]: Done.

File: 08/README.md (hunk 2 above)
Comment (mwhittaker) [reply 08/README.md]: Done.

**Pull request merged**